python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -691,7 +691,8 @@ def _flush(self, objs):
 
         for obj in objs:
             # check if the object is dirty, if not do nothing
-            if not obj._dirty:
+            this_id = self.find(obj)
+            if not obj._dirty or this_id not in self._loaded_ids:
```

Comment:
I'm concerned by it but I've not been able to track down a solid example so I'm in favor of always having the correct behavior here at least. I think we can end up with a partly-loaded job being dirty but not marked as correctly loaded (this I'm not 100% sure about but I suspect).
This is more of an attempt to stave off future problems.